File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        diffMap = dict()

        for i in range(len(nums)):
            difference = target - nums[i]

            if difference in diffMap:
                return [diffMap[difference], i]
            
            diffMap[nums[i]] = i
        

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.js
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers, target) {
        let leftPointer = 0;
        let rightPointer = numbers.length - 1;

        while (leftPointer < rightPointer) {
            let sum = numbers[leftPointer] + numbers[rightPointer];

            if (sum > target) {
                rightPointer--;
            } else if (sum < target) {
                leftPointer++;
            } else {
                return [leftPointer + 1, rightPointer + 1];
            }
        }
    }
}
